src/iter/mod.rs

```diff
@@ -10,12 +11,11 @@ use std::sync::atomic::Ordering;
 #[derive(Debug)]
 pub struct Iter<'g, K, V> {
     pub(crate) node_iter: NodeIter<'g, K, V>,
-    pub(crate) guard: &'g Guard,
+    pub(crate) guard: &'g Guard<'g>,
 }
 
-impl<'g, K, V> Iterator for Iter<'g, K, V> {
-    type Item = (&'g K, &'g V);
-    fn next(&mut self) -> Option<Self::Item> {
+impl<'g, K, V> Iter<'g, K, V> {
+    pub(crate) fn next_internal(&mut self) -> Option<(&'g K, &'g Linked<V>)> {
```

Comment:
As far as I can tell the only use of this is in `impl Iterator for Iter`, so why separate it out?

Reply:
It's used in `retain`, we need the `&Linked<V>` instead of `&V` in order to `retire`.

Reply:
Ah, I missed that. All good then!